be1-go/hub/standard_hub/hub_state/ids_by_channel.go

```diff
@@ -0,0 +1,59 @@
+package hub_state
+
+import (
+	"golang.org/x/exp/slices"
+	"maps"
+	"sync"
+)
+
+// IdsByChannel provides a thread-safe structure that stores a channel id with its corresponding message ids
+type IdsByChannel struct {
+	sync.RWMutex
+	table map[string][]string
+}
+
+// NewIdsByChannel creates a new IdsByChannel structure
+func NewIdsByChannel() IdsByChannel {
+	return IdsByChannel{
+		table: make(map[string][]string),
+	}
+}
+
+// Add adds a message id to the table
+func (i *IdsByChannel) Add(channel string, id string) {
```

Comment:
Conveniently, using type embedding would allow you to override the `Add` of the underlying ThreadSafeMap to put this extra logic.